### PR TITLE
Include timestamp in PersistentRepr and EventEnvelope

### PR DESCRIPTION
**Pull request by renatocaval**

### CODE REVIEW

File: src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala [hunk 1]
```diff
@@ -86,8 +86,18 @@ class JdbcAsyncWriteJournal(config: Config) extends AsyncWriteJournal {
...
-    val persistenceId = messages.head.persistenceId
+
+    // add timestamp to all payloads in all AtomicWrite messages
+    val timedMessages =
+      messages.map { atomWrt =>
+        // since they are all persisted atomically,
+        // all PersistentRepr on the same atomic batch gets the same timestamp
+        val now = System.currentTimeMillis()
+        atomWrt.copy(payload = atomWrt.payload.map(pr => pr.withTimestamp(now)))
+      }
```
Comment (renatocaval) [inline src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala]: I was in doubt where to add this. 

Initially, I thought that this should be added by Akka itself when internally creating the `Seq[AtomicWrite]`.

The problem is that the sequence is created using an internal mutable var so each `PersistenceRepr` would get a different timestamp. Not such a big issue though. 

Another reason is, depending on the plugin, this timestamp could differ from the timestamp that is added to the envelope.  For instance, Cassandra plugin will use the Offset, so filling it by default is useless (or even confusing when debugging). 

I guess each plugin implementation should decide when to fill it and how to build the `EventEnvelope`. 

Anyway, adding it here like that for the moment. We can discuss if it's worth or more 'correct' to let Akka fill this field.

File: src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala [hunk 2]
```diff
@@ -171,7 +171,7 @@ class JdbcReadJournal(config: Config, configPath: String)(implicit val system: E
     }
       .mapConcat(identity)
       .mapConcat(adaptEvents)
-      .map(repr => EventEnvelope(Sequence(repr.sequenceNr), repr.persistenceId, repr.sequenceNr, repr.payload))
+      .map(repr => EventEnvelope(Sequence(repr.sequenceNr), repr.persistenceId, repr.sequenceNr, repr.payload, repr.timestamp))
```
Comment (renatocaval) [inline src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala]: It's missing test for this code path. I will add it.

Comment (renatocaval) [reply src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala]: done

File: src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala (hunk 1 above)
Comment (patriknw) [reply src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala]: `asyncWriteMessages` is the right place.
The reason for not setting it in Akka itself is that journals will anyway store it in different ways and some (like Cassandra) may already have a timestamp. Also, serializing the PersistenceRepr is not the recommended way, so the journal will typically have a separate field for the timestamp (but that is for later in the jdbc plugin).

**Pull request merged**